src/imap/mod.rs

```diff
@@ -591,6 +670,7 @@ impl Imap {
 
         self.set_config_last_seen_uid(context, &folder, new_uid_validity, new_last_seen_uid)
             .await;
+        job::schedule_resync(context).await;
```

Comment:
So this does a full resync on every select folder?  That seems... expensive?

(I may also have misunderstood this)

Reply:
There is a early exit about for the case when `new_uid_validity == uid_validity`. This code handles UID validity changes.